### PR TITLE
Update plan-build default_strip behavior

### LINKED ISSUE #6186: hab-sup 0.74.0 binary size is 20x larger than 0.73.0
The hab-sup package has grown substantially between 0.73.0 and 0.74.0:

```
-rw-r--r-- 1 root root 3.6M Dec  7 09:51 /hab/cache/artifacts/core-hab-sup-0.70.0-20181205195836-x86_64-linux.hart
-rw-r--r-- 1 root root 3.7M Feb 20 18:13 /hab/cache/artifacts/core-hab-sup-0.73.0-20190115010906-x86_64-linux.hart
-rw-r--r-- 1 root root  44M Feb 12 13:13 /hab/cache/artifacts/core-hab-sup-0.74.0-20190204225911-x86_64-linux.hart
```

Unpacked, the hab-sup binary is 20x larger, now over 300M:

```
# ls -lh /hab/pkgs/core/hab-sup/0.73.0/20190115010906/bin/hab-sup 
-rwxr-xr-x 1 root root 15M Jan 15 01:21 /hab/pkgs/core/hab-sup/0.73.0/20190115010906/bin/hab-sup
# ls -lh /hab/pkgs/core/hab-sup/0.74.0/20190204225911/bin/hab-sup 
-rwxr-xr-x 1 root root 303M Feb  4 23:11 /hab/pkgs/core/hab-sup/0.74.0/20190204225911/bin/hab-sup
```

This size increase substantially increases the download and unpack time for this habitat supervisor.

It appears that debug info is driving this size increase:

```
# file /hab/pkgs/core/hab-sup/0.73.0/20190115010906/bin/hab-sup 
/hab/pkgs/core/hab-sup/0.73.0/20190115010906/bin/hab-sup: ELF 64-bit LSB shared object, x86-64, version 1 (SYSV), dynamically linked, interpreter /hab/pkgs/core/glibc/2.27/20180608041157/lib/ld-linux-x86-64.so.2, for GNU/Linux 3.2.0, stripped
# file /hab/pkgs/core/hab-sup/0.74.0/20190204225911/bin/hab-sup 
/hab/pkgs/core/hab-sup/0.74.0/20190204225911/bin/hab-sup: ELF 64-bit LSB shared object, x86-64, version 1 (SYSV), dynamically linked, interpreter /hab/pkgs/core/glibc/2.27/20190115002733/lib/ld-linux-x86-64.so.2, for GNU/Linux 3.2.0, with debug_info, not stripped
```

```
# size -A -d /hab/pkgs/core/hab-sup/0.74.0/20190204225911/bin/hab-sup   
/hab/pkgs/core/hab-sup/0.74.0/20190204225911/bin/hab-sup  :
section                   size       addr
.interp                     66        736
.note.ABI-tag               32        804
.hash                     3836        840
.gnu.hash                  444       4680
.dynsym                   8040       5128
.dynstr                   5946      13168
.gnu.version               670      19114
.gnu.version_r             464      19784
.rela.dyn               926880      20248
.rela.plt                  744     947128
.init                       23     950272
.plt                       512     950304
.plt.got                   288     950816
.text                 11238128     951104
.fini                        9   12189232
.rodata                2005020   12189696
.debug_gdb_scripts          34   14194716
.eh_frame_hdr           250036   14194752
.eh_frame              1491656   14444792
.gcc_except_table       688976   15936448
.tdata                   10896   16630592
.tbss                     1088   16641504
.init_array                 16   16641504
.fini_array                  8   16641520
.data.rel.ro            596800   16641536
.dynamic                   704   17238336
.got                     29688   17239040
.data                      818   17268736
.bss                   2162712   17269568
.comment                    62          0
.debug_aranges           14880          0
.debug_pubnames       22529162          0
.debug_info           92871715          0
.debug_abbrev          1305914          0
.debug_line           17304113          0
.debug_frame              4696          0
.debug_str            28000300          0
.debug_loc            76054417          0
.debug_macinfo            1724          0
.debug_pubtypes       29056180          0
.debug_ranges         28560320          0
.debug_macro            125634          0
Total                315253651
```

I am generally in favor of debuggable binaries even in production; but this feels like a high cost to pay and that there might be some middel ground.

### PR DESCRIPTION
**Pull request by smacfarlane**
Closes #6186 

In the recent [core-plans refresh](https://github.com/habitat-sh/core-plans/pull/1767) we updated `file` from 5.32 to 5.34.  In between those two releases was a fix to the `file` command to correctly report modern PIE executables as `application/x-pie-executable` instead of the previously incorrect `application/x-sharedlib`.    

https://gcc.gnu.org/bugzilla/show_bug.cgi?id=81498
https://bugs.launchpad.net/ubuntu/+source/file/+bug/1747711

This resulted files that were previously misreported as `x-sharedlib` to no longer match a [branch of the case statement](https://github.com/habitat-sh/habitat/blob/0.75.0/components/plan-build/bin/hab-plan-build.sh#L2053-L2057) in `do_default_strip` and were therefore no longer stripped of debug symbols leading to the bloated file sizes reported in #6186.

This PR adds an additional branch to the case statement in do_default_strip, to give `application/x-pie-executable` the previous `strip --strip-unneeded` behavior of `application/x-sharedlib`.  While the original intent was to `strip --strip-all` for executable files,  that hasn't been the behavior for an unknown period of time, possibly the lifetime of the project.  While `--strip-all` is probably the desired behavior, I think we should introduce that change after evaluating the impact across the ecosystem.  

Building the Habitat supervisor with these changes results in a reduction in file size in line with previous values:

```
3816853 core-hab-sup-0.73.0-20190115010906-x86_64-linux.hart
46022649 core-hab-sup-0.74.0-20190204225911-x86_64-linux.hart
43047381 core-hab-sup-0.75.0-20190220000453-x86_64-linux.hart
3937420 smacfarlane-hab-sup-0.76.0-dev-20190220210136-x86_64-linux.hart
```

Signed-off-by: Scott Macfarlane <smacfarlane@chef.io>

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

File: components/plan-build/bin/hab-plan-build.sh [hunk 1]
```diff
@@ -2046,12 +2046,21 @@ do_strip() {
 }
 
 # Default implementation for the `do_strip()` phase.
+# TODO(SM): Previous versions of the `find` utility reported x-pie-exectuable
```
Comment (baumanj) [inline components/plan-build/bin/hab-plan-build.sh]:
```suggestion
# TODO(SM): Previous versions of the `file` utility reported x-pie-exectuable
```

File: components/plan-build/bin/hab-plan-build.sh [hunk 2]
```diff
@@ -2046,12 +2046,21 @@ do_strip() {
...
+# as x-sharedlib. This means that while the intent was to `--strip-all` for 
+# x-executable, in reality we have been running `--strip-unneeded`. In order to 
+# be consistant with past behavior we will pass `--strip-unneeded` when stripping
+# x-pie-executable. In the future, we will need to make a decision as to the behavior
```
Comment (baumanj) [inline components/plan-build/bin/hab-plan-build.sh]: Do we have an issue filed for this future work? There are some other fix-ups that could make this code a bit more robust.

Comment (smacfarlane) [reply components/plan-build/bin/hab-plan-build.sh]: Yes, I added #6191 to track the work

**Pull request merged**